### PR TITLE
GOAL2-653 deprecate IsRelay field in NodeConfig

### PR DESCRIPTION
**Pull request by algobolson**
## Summary

Simplify NodeConfig struct. IsRelay is redundant, IsRelay() = nc.NetAddress != ""

## Test Plan

compilation plus unit tests

### CODE REVIEW

File: cmd/netgoal/generate.go [hunk 1]
```diff
@@ -199,7 +199,10 @@ func generateNetworkTemplate(templateFilename string, wallets, relays, nodeHosts
 		newNode.NodeNameMatchRegex = ""
 		newNode.FractionApply = 0.0
 		newNode.Name = name
-		newNode.IsRelay = true
+		if newNode.NetAddress == "" {
```
Comment (tsachiherman) [inline cmd/netgoal/generate.go]: I think that you don't want to have these workarounds; you want to make sure that the relayTemplates is configured correctly. If not, we need to fix it there.

Comment (algobolson) [reply cmd/netgoal/generate.go]: yeah, the other option would to be to fail with error message if a "relay" template failed to set NetAddress. I'll work that up.

**Comment by algobolson** [comment]
looks like test/testdata/nettemplates/*.json need fixing; didn't show up when grepping the .go sources.

**Comment by algobolson** [comment]
There's a separate `IsRelay bool` in a different template system in `netdeploy/networkTemplate.go`
That's what interprets the templates in `test/testdata/nettemplates/*.json`
That structure doesn't even have a 'NetAddress' field because it's a simple ephemeral system I think only used by `goal network create` which makes local private networks, where all the NetAddress values are effectively "127.0.0.1:0" so that they only listen locally and get a random dynamic port each time they start up.
I'm tempted to leave that system alone.

File: cmd/netgoal/generate.go (hunk 1 above)
Comment (Karmastic) [reply cmd/netgoal/generate.go]: Yeah - fail if netaddress not specified for relays - don't default.

File: netdeploy/remote/nodeConfig.go [hunk 2]
```diff
@@ -19,7 +19,7 @@ package remote
 // NodeConfig represents the configuration settings to apply to a single node running on a host
 type NodeConfig struct {
 	Name               string `json:",omitempty"`
-	IsRelay            bool
+	IsRelayDEPRECATED  bool   `json:"IsRelay,omitempty"` // The field still exists to parse any old serialized NodeConfig objects, but "IsRelay" should not be used.
```
[the remark this answers is not in the document]
Comment (Karmastic) [reply netdeploy/remote/nodeConfig.go]: Yeah - for most things we don't fail on unknown properties (we do on things like genesis.json)

File: netdeploy/remote/nodecfg/nodeDir.go [hunk 3]
```diff
@@ -143,7 +143,7 @@ func (nd *nodeDir) configureNetAddress(address string) (err error) {
 	}
 	fmt.Fprintf(os.Stdout, " - Assigning NetAddress: %s\n", address)
 	nd.config.NetAddress = address
-	if address != "" && address[0] == ':' && nd.IsRelay {
+	if address != "" && address[0] == ':' && nd.IsRelay() {
```
Comment (Karmastic) [inline netdeploy/remote/nodecfg/nodeDir.go]: You can remove the call to nd.IsRelay() - it's redundant now.

File: netdeploy/remote/nodeConfig.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (algobolson) [reply netdeploy/remote/nodeConfig.go]: field is deleted now

File: cmd/netgoal/generate.go (hunk 1 above)
Comment (algobolson) [reply cmd/netgoal/generate.go]: returns an error now

File: netdeploy/remote/nodecfg/nodeDir.go (hunk 3 above)
Comment (algobolson) [reply netdeploy/remote/nodecfg/nodeDir.go]: redundancy eliminated, but kept nd.IsRelay() because that has some meaning

**Pull request merged**